### ISSUE TITLE
[Feature] Remove ads from rss feed ?

### ISSUE THREAD

**Issue by typhoe** [FreshRSS/FreshRSS#3598]
Hi,

in some feeds, I have ads that only appear when loading the page from an aggregator because they are not detected anymore by my adblocker.

Ex: 
In this feed: https://korben.info/feed

Let's open the latest article: https://korben.info/comment-contourner-censure-snowflakes.html
ublock origin will block the ad for nordvpn:
```
<div id="dEbO2mTU-2003902640" data-cfptl="1" data-cfpa="122880">
<div class="wp-block-group alignwide bg-post bg-full p-10"><div class="wp-block-group__inner-container">
<h2><a href="http://bit.ly/3nBnyM3helio?source=korben.info" target="_blank" rel="noreferrer noopener">En plus du VPN, vous avez Nordpass, le nouveau gestionnaire de mot de passe à -70% </a></h2>
<figure class="wp-block-image size-large is-resized"><img data-attachment-id="122877" data-permalink="https://korben.info/?attachment_id=122877" data-orig-file="https://korben.info/app/uploads/2021/04/nordpass.png" data-orig-size="973,551" data-comments-opened="1" data-image-meta="{&quot;aperture&quot;:&quot;0&quot;,&quot;credit&quot;:&quot;&quot;,&quot;camera&quot;:&quot;&quot;,&quot;caption&quot;:&quot;&quot;,&quot;created_timestamp&quot;:&quot;0&quot;,&quot;copyright&quot;:&quot;&quot;,&quot;focal_length&quot;:&quot;0&quot;,&quot;iso&quot;:&quot;0&quot;,&quot;shutter_speed&quot;:&quot;0&quot;,&quot;title&quot;:&quot;&quot;,&quot;orientation&quot;:&quot;0&quot;}" data-image-title="nordpass" data-image-description="" data-medium-file="https://korben.info/app/uploads/2021/04/nordpass-300x170.png" data-large-file="https://korben.info/app/uploads/2021/04/nordpass.png" loading="lazy" src="https://korben.info/app/uploads/2021/04/nordpass.png" alt="" class="wp-image-122877" width="688" height="389" srcset="https://korben.info/app/uploads/2021/04/nordpass.png 973w, https://korben.info/app/uploads/2021/04/nordpass-300x170.png 300w, https://korben.info/app/uploads/2021/04/nordpass-768x435.png 768w" sizes="(max-width: 688px) 100vw, 688px"></figure>
<p>NordPass est compatible avec les principaux navigateurs: Google Chrome, Mozilla Firefox, Apple Safari. Opera et Microsoft Edge.</p>
<p><strong>Simplicité</strong><br><a href="http://bit.ly/3nBnyM3helio?source=korben.info" target="_blank" rel="noreferrer noopener">NordPass</a> reconnaît vos sites Web préférés et remplit automatiquement vos informations de connexion lors de la connexion.<br>Faites-vous des achats en ligne ? Stockez les détails de votre carte de crédit et les informations d’expédition pour un paiement plus fluide. C’est smooth, c’est fun c’est sans prise de tête.</p>
<p>Actuellement, le gestionnaire propose -70% sur le prix de l’abonnement.</p>
<div class="wp-block-buttons">
<div class="wp-block-button is-style-fill"><a class="wp-block-button__link has-kbn-primary-background-color has-background" href="http://bit.ly/3nBnyM3helio" target="_blank" rel="noreferrer noopener">Tester Nordpass</a></div>
</div>
</div></div>
</div>
```

But when I display the same article from FreshRSS, the ad will be displayed (as it is in the content of the feed).

Is there a way to block it or even better, to remove it from the feed content?

I tried to find a solution using the selectors but with no success.

Thank you.

**Comment by martinrotter** [comment]
I use FreshRSS with RSS Guard (which I am author of) and I block certain elements with message [filters](https://github.com/martinrotter/rssguard/blob/master/resources/docs/Message-filters.md) and [AdBlock](https://github.com/martinrotter/rssguard/blob/master/resources/docs/Documentation.md#adblock).

**Comment by typhoe** [comment]
Hi,

thanks for the reply and suggestion!

I'm indeed looking into doing the same right now: Finding the right filter used to prevent the add on the origin site and applying it also on my FreshRSS url now.
That seems to be what you do on RSS Guard Web (and I won't use the windows app, I'm running on linux ;-) ).

**Comment by martinrotter** [comment]
> 
> 
> Hi,
> 
> thanks for the reply and suggestion!
> 
> I'm indeed looking into doing the same right now: Finding the right filter used to prevent the add on the origin site and applying it also on my FreshRSS url now.
> That seems to be what you do on RSS Guard Web (and I won't use the windows app, I'm running on linux ;-) ).

RSS Guard is desktop app with filtering and included AdBlock. It runs on Linux too of course (it is cross platform).

**Comment by typhoe** [comment]
Oh, nice to know! (my mistake :-p )
Will try then, thanks!

**Comment by martinrotter** [comment]
OK, I haven't done any testing lately on Linux, so there might be some bugs, preferably use development build AppImage, which contains some nice latest fixes and features. Especially for FreshRSS plugin.

**Comment by rom-1** [comment]
I just want to back this Feature Request. I have also the very same problem with some of my feeds. Using an adblocker/3rd party solution is in my opinion only a bad workaround to solve this problem. FreshRSS has the ability to load article content using Article CSS selectors, and it would be great if users could utilize them for substracting/ignoring specific tags. 
(btw: tiny tiny rss has a powerful extension called [feediron](https://github.com/feediron/ttrss_plugin-feediron) which does exactly this)

Example feed: https://www.heise.de/rss/heise-atom.xml
Currently using CSS selectors: #meldung .article-content, #meldung p.a-article-header__lead, #meldung .article-image

--> the feed content still contains Ads/images which are inserted within .article-content

**Comment by math-GH** [comment]
feediron's description:
`It allows you to replace an article's contents by the contents of an element on the linked URL's page`

FreshRSS can do it without an extension. In the feed settings you will find "Article CSS selector on original website"
https://freshrss.github.io/FreshRSS/en/users/04_Subscriptions.html

**Comment by rom-1** [comment]
no, feediron is much more powerful, as you can strip content from the url. if you have a 
`<div id="ads"> <img src=ad.png> </div>`
in the original page, you can use feediron to simply strip/ignore that specific div. This is currently not possible with freshrss.

**Comment by helmut72** [comment]
+1! Heavily missing something like [Feediron](https://github.com/feediron/ttrss_plugin-feediron) or [Scraper rules from Miniflux2](https://miniflux.app/docs/rules.html).

One of the last few piece to have a nearly perfect RSS server. But it looks like it's coming https://github.com/FreshRSS/FreshRSS/pull/4220

**Comment by math-GH** [comment]
@helmut72 : Check #4175
It will make it possible to use CSS to hide the advertisements.

**Comment by helmut72** [comment]
@math-GH Thanks! Looks good but unfortunately I still need to test it. Had and will have some busy days.

**Comment by Alkarex** [comment]
https://github.com/FreshRSS/FreshRSS/pull/4501

**Comment by m00nwtchr** [comment]
I'd be nice if there was an option which allows removal of specific content from the body of the RSS feed itself (ie without having to make freshrss fetch the article from the site)
